annotationProcessor/src/main/java/org/corfudb/annotations/ObjectAnnotationProcessor.java

```diff
@@ -385,18 +385,14 @@ public void generateProxy(TypeElement classElement)
                             }
                         } else if (method.getAnnotation(InterfaceOverride.class) != null) {
                             interfacesToAdd.add(ParameterizedTypeName.get(ifaceElement.asType()));
-                            if (overwrite.isPresent()) {
-                                methodSet.remove(overwrite.get());
-                            }
-                            methodSet.add(new SmrMethodInfo(method,
-                                        ifaceElement));
+                            overwrite.ifPresent(methodSet::remove);
```

Comment:
It doesn't seem to be the previous behavior where we were passing the content of overwrite (if present) to be removed from the set. method references in lambda apply on the value presented to them which is boolean in your statement not the content of overwrite.

Reply:
Intellij Idea just highlights this code and offers to replace to ifPresent.
Anyway, I won't bring this change in the other PR

Reply:
Intellij sometimes suggest equivalent syntax changes that might lead to sematic differences. I think Intellij thought that the method (ifPresent) returns the actual value while in reality it returns a boolean (not useful for the method reference). But you can check more, I might be wrong.